include/seastar/coroutine/exception.hh

```diff
@@ -30,7 +30,7 @@ namespace internal {
 struct exception_awaiter {
     std::exception_ptr eptr;
 
-    explicit exception_awaiter(std::exception_ptr&& eptr) : eptr(std::move(eptr)) {}
+    explicit exception_awaiter(std::exception_ptr&& eptr) noexcept : eptr(std::move(eptr)) {}
```

Comment:
I don't understand what's an untyped exception_ptr.

Reply:
The symptom is that the boost testing framework reports the following:
```
unknown location(0): fatal error: in "sstable_directory_test_table_scan_invalid_file": unknown type
```

I guess that the type of the held object is `std::exception_ptr` (rather than untyped as I said)